docs/source/best_practices.rst

```diff
@@ -0,0 +1,222 @@
+.. _best_practices:
+
+Best Practices and Performance Tips
+===================================
+
+Sort and Use Read Filtering
+---------------------------
+
+If you often read by key ranges or perform lots of logic with groups of related records, you should consider using Dask Dataframe's ``shuffle``.
+This operation ensures that all rows of a given key will be within a single partition.
+This is helpful for querying records on a specific key or keys such as customer IDs or session keys, as it allows Dask to skip partitions based on the partition min and max values thus avoiding reading each record.
+This can save a large amount of IO time and is especially helpful when using a network file system.
+
+For example, querying a specific pickup time from a taxi dataset ends up returning a result with over 200 partitions as each of these partitions needs to be checked for that key.
+.. code-block:: python
+    ddf = dd.read_parquet('/data/taxi_pq_2GB', split_row_groups=False)
+    c.create_table('taxi_unsorted', ddf)
+    c.sql("select * from taxi_unsorted where DAYOFMONTH(pickup_datetime) = 15").npartitions
+
+.. code-block::
+    244
+
+But, if you were to instead sort by the pickup time and use the ``DISTRIBUTE BY`` operation, which is equivalent to Dask Dataframe's shuffle, you can reduce the number of partitions in the result to 1.
+
+.. code-block:: python
+    def intra_partition_sort(df, sort_keys):
+        return df.sort_values(sort_keys)
+
+    c.sql("""
+    SELECT
+        DAYOFMONTH(pickup_datetime) AS dom,
+        HOUR(pickup_datetime) AS hr,
+        *
+    FROM
+        taxi_unsorted
+    DISTRIBUTE BY dom
+    """).map_partitions(intra_partition_sort, ['dom', 'hr']).to_parquet('/data/taxi_sorted')
+
+.. code-block:: python
+    sorted_ddf = dd.read_parquet(
+        '/data/taxi_sorted',
+        split_row_groups=False,
+        filters=[
+            [("dom", "==", 15)]
+        ]
+    )
+
+    c.create_table("taxi_sorted", sorted_ddf)
+    .sql("SELECT * FROM taxi_sorted WHERE dom = 15").npartitions
+
+.. code-block::
+    1
+
+This comes with a large corresponding boost in computation speed. For example,
+
+.. code-block:: python
```

Comment:
This (and other code blocks) doesn't display on https://github.com/dask-contrib/dask-sql/blob/55237bce95ca80ff582420e140acc70acad04137/docs/source/best_practices.rst ?

Reply:
Should add to list here: https://github.com/dask-contrib/dask-sql/blob/5e1b40769b5281906a667d4c45c25bc39fb18fbc/docs/source/index.rst